tests/acceptance/ENFormCest.php

```diff
@@ -301,90 +297,257 @@ public function createAnEnBlock( AcceptanceTester $I ) {
-		$I->selectOption( ENBlock::$enPageSelect, $option );
-		$shortcode_attributes[ ENBlock::$enPageSelect ] = $option;
+		$page_id = $I->havePageInDatabase([
+			'post_name'    => $slug,
+			'post_status'  => 'publish',
+			'post_content' => $generated_enformblock
+		]);
+
+		// Navigate to the newly created page
+		$I->amOnPage('/' . $slug);
+
+		// Create an array with the engaging networks form values to assert the en api request payload later in the test.
+		$user_form_data = [
+			'standardFieldNames' => true,
+		];
+
+
+		// Fill the form's questions/checkboxes.
+		foreach ( $this->form_fields['questions'] as $form_field ) {
+			$field_name = $form_field['name'];
+			$field_type = $this->form_fields_attributes[ $field_name ]['type'];
+
+			if ( in_array( $field_type, [ 'Email', 'Text' ] ) ) {
+				$I->fillField( 'supporter.questions.' . $form_field['id'], 'Lorem ipsum' );
+				$user_form_data['supporter']['questions'][ 'question.' . $form_field['id'] ] = 'Lorem ipsum';
+			} else if ( 'Checkbox' === $field_type ) {
+				$user_form_data['supporter']['questions'][ 'question.' . $form_field['id'] ] = 'Y';
+				$I->click('.en__field__element--check');
+			}
+		}
+
+		// Fill the form's fields
+		foreach ( $this->form_fields['fields'] as $form_field ) {
+			$field_name  = $form_field['name'];
+			$field_type  = $this->form_fields_attributes[ $field_name ]['type'];
+			$field_value = $this->form_fields_attributes[ $field_name ]['default_value'];
+			if ( in_array( $field_type, [ 'Email', 'Text' ] ) ) {
+
+				$I->fillField( 'supporter.' . $form_field['property'], 'Lorem ipsum' );
+				$user_form_data['supporter'][ $form_field['property'] ] = 'Lorem ipsum';
+			} else if ( 'Hidden' === $field_type ) {
+
+				$I->seeElementInDOM( 'input',
+					[
+						'name'  => 'supporter.' . $form_field['property'],
+						'value' => $field_value,
+					]
+				);
+				$user_form_data['supporter'][ $form_field['property'] ] = $field_value;
+			}
+		}
+
+		$I->click( '#p4en_form_save_button' );
+
+		// Check email validation.
+		$I->see( 'Please enter a valid e-mail address.' );
+
+		// Fill a correct email to continue with the form submit.
+		$I->fillField( 'supporter.emailAddress', 'test@example.com' );
+		$user_form_data['supporter']['emailAddress'] = 'test@example.com';
+
+		$json_body = $I->executeJS( 'return JSON.stringify(p4_enform_frontend.getFormData());' );
+		$I->clickWithLeftButton( '.btn-primary' );
+
+		// Assert that the generated json payload contains all fields.
+		$I->assertEquals( json_decode( $json_body, true ), $user_form_data );
+	}
+
+	/**
+	 * Create a WordPress page and add an enblock(full-width-bg) in it and check the enform on frontend page.
+	 *
+	 * @depends createAnEnForm
+	 * @group enform
+	 * @group engaging-networks
+	 *
+	 * @param AcceptanceTester $I
+	 *
+	 * @throws \Codeception\Exception\ModuleException
+	 */
+	public function createAnTestEnBlockFullWidthBG( AcceptanceTester $I ) {
```

Comment:
Same here: not a huge deal, but I think there's a typo in the function name here? Shouldn't it be `createAndTestEnBlockFullWidthBG`?